### PR TITLE
[16.0][ADD] pos_bypass_global_discount

### PR DESCRIPTION
**Pull request by trisdoan**
### Note
- This module is meant to be installed as a companion to `sale_global_discount`. It makes sure the POS will also honors the `bypass_global_discount` field when user inputs a Discount at the Order level.

### CODE REVIEW

**Review by cyrilmanuel — APPROVED** [review]

**Comment by cyrilmanuel** [comment]
adding description on the PR : 

This module is meant to be installed as a companion to `sale_global_discount`. It makes sure the POS will also honors the `bypass_global_discount` field when user inputs a Discount at the Order level.

File: pos_bypass_global_discount/models/pos_session.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import models
+
+
+class POSSession(models.Model):
+    _inherit = "pos.session"
+
+    def _loader_params_product_product(self):
+        params = super()._loader_params_product_product()
+        params["search_params"]["fields"].append("bypass_global_discount")
```
Comment (Ricardoalso) [inline pos_bypass_global_discount/models/pos_session.py]:
```suggestion
if "search_params" in params and "fields" in params["search_params"]:
  params["search_params"]["fields"].append("bypass_global_discount")
```

**Review by Ricardoalso — APPROVED** [review]

**Review by ivantodorovich — APPROVED** [review]

**Comment by simahawk** [comment]
https://github.com/OCA/pos/issues/1236

**Pull request merged**